kafka/src/main/java/io/specmesh/kafka/Provisioner.java

```diff
@@ -176,17 +212,24 @@ static ParsedSchema getSchema(
      * @param schemaResources schema path
      * @param adminClient kafka admin client
      * @param schemaRegistryClient sr client
+     * @return status of provisioning
      * @throws ProvisioningException when cant provision resources
      */
-    public static void provision(
+    public static Status provision(
             final KafkaApiSpec apiSpec,
             final String schemaResources,
             final Admin adminClient,
             final Optional<SchemaRegistryClient> schemaRegistryClient) {
-        provisionTopics(apiSpec, adminClient);
+        final var topicProvisionStatus = provisionTopics(apiSpec, adminClient);
+        final var status = Status.builder().topics(topicProvisionStatus);
         schemaRegistryClient.ifPresent(
-                registryClient -> provisionSchemas(apiSpec, schemaResources, registryClient));
-        provisionAcls(apiSpec, adminClient);
+                registryClient ->
+                        status.schemas(
+                                Optional.of(
+                                        provisionSchemas(
+                                                apiSpec, schemaResources, registryClient))));
```

Comment:
Why take an `Optional` on a builder method? Just don't call the method if you don't want it!